src/transaction/declare_v2.rs

```diff
@@ -458,7 +458,13 @@ impl DeclareV2 {
             ));
         }
         if let Some(ref class) = self.sierra_contract_class {
-            state.set_sierra_program(&self.sierra_class_hash, class.sierra_program.clone())?;
+            state.set_sierra_program(
```

Comment:
we should check if we want to do this without the cairo-native feature.

Reply:
What do you mean? Gating the entire `if` block within `#[cfg(feature = "cairo-native")]`?